src/udp_wrap.cc

```diff
@@ -464,14 +505,14 @@ void UDPWrap::DoSend(const FunctionCallbackInfo<Value>& args, int family) {
     CHECK(args[3]->IsBoolean());
   }
 
-  Local<Object> req_wrap_obj = args[0].As<Object>();
   Local<Array> chunks = args[1].As<Array>();
   // it is faster to fetch the length of the
   // array in js-land
   size_t count = args[2].As<Uint32>()->Value();
-  const bool have_callback = sendto ? args[5]->IsTrue() : args[3]->IsTrue();
 
-  size_t msg_size = 0;
+  wrap->current_send_req_wrap_ = args[0].As<Object>();
+  wrap->current_send_has_callback_ =
+      sendto ? args[5]->IsTrue() : args[3]->IsTrue();
```

Comment:
Looks like those are used to pass up additional data to `CreateSendWrap()`? Inelegant..

At the very least, they should be moved to right before the `Send()` call (that's what may or may not call `CreateSendWrap()`, right?) and cleared right after.

Reply:
I think that should work, yes.